src/pkg/utils/network.go

```diff
@@ -69,27 +68,74 @@ func Fetch(url string) io.ReadCloser {
 	return resp.Body
 }
 
+func parseChecksum(src string) (string, string, error) {
+	atSymbolCount := strings.Count(src, "@")
+	var checksum string
+	if atSymbolCount > 0 {
+		parsed, err := url.Parse(src)
+		if err != nil {
+			return src, checksum, fmt.Errorf("unable to parse the URL: %s", src)
+		}
+		if atSymbolCount == 1 && parsed.User != nil {
+			return src, checksum, nil
+		}
+
+		index := strings.LastIndex(src, "@")
+		checksum = src[index+1:]
+		src = src[:index]
+	}
+	return src, checksum, nil
```

Comment:
Also I think this will fail with a url that uses an @ but doesn't have a shasum.
i.e. `https://user:pass@hello.world?foo=bar`

Reply:
`parsed.User != nil` handles that